.github/workflows/cd.yml

```diff
@@ -22,28 +22,39 @@ on:
         type: string
         required: False
 
-env:
-  API_VERSION: dev
-  WORKER_VERSION: latest
-  CLI_VERSION: latest
 
 jobs:
   set-component-versions:
     runs-on: ubuntu-latest
+    outputs:
+      api_version: ${{ steps.api_version.outputs.result }}
+      worker_version: ${{ steps.worker_version.outputs.result }}
+      cli_version: ${{ steps.cli_version.outputs.result }}
+
     steps:
-    - name: use custom if workflow_dispatch
+
+    - id: api_version
+      name: dynamic input api version
+      run: |
+        if [[ ${{ inputs.api_version }} == "" ]]; then echo dev; else echo ${{ inputs.api_version }}; fi
+
+    - id: worker_version
+      name: dynamic input worker version
+      run: |
+        if [[ ${{ inputs.worker_version }} == "" ]]; then echo dev; else echo ${{ inputs.worker_version }}; fi
+
+    - id: cli_version
+      name: dynamic input cli version
       run: |
-        if [[ ${{ inputs.api_version }} == "" ]]; then echo "API_VERSION=${{ inputs.api_version }}" >> $GITHUB_ENV; fi
-        if [[ ${{ inputs.worker_version }}  == "" ]]; then echo "WORKER_VERSION=${{ inputs.worker_version }}" >> $GITHUB_ENV; fi
-        if [[ ${{ inputs.cli_version }}  == "" ]]; then echo "CLI_VERSION=${{ inputs.cli_version }}" >> $GITHUB_ENV; fi
+        if [[ ${{ inputs.cli_version }} == "" ]]; then echo dev; else echo ${{ inputs.cli_version }}; fi
```

Comment:
I am wondering if `inputs.api_version` is not provided are we sure it will be equal to `""`?
Are we sure it will exist at all and there won't be an error when you try to access an attribute that doesn't exist?

Reply:
Yes, it is correct. I've tested here.
https://github.com/vmware/repository-service-tuf-api/actions/runs/4242147999/jobs/7373275082

Reply:
This is not the latest commit.
Best hide this comment.